src/app/Processor/ProcessorEdit/ProcessorEdit.tsx

```diff
@@ -404,14 +411,33 @@ const ProcessorEdit = (props: ProcessorEditProps): JSX.Element => {
                               </TextContent>
                               <CodeEditor
                                 id={"transformation-template"}
+                                className={css(
+                                  "processor-edit__transformation-template",
+                                  malformedTransformationTemplate
+                                    ? "processor-field-error"
+                                    : ""
+                                )}
                                 height={"300px"}
                                 isLineNumbersVisible={true}
                                 code={transformation}
-                                onChange={setTransformation}
+                                onChange={(value): void => {
+                                  setTransformation(value);
+                                  resetTransformationStatus?.();
```

Comment:
I was about to start reviewing but then I stopped to think about this, clearing the TT error on template change.

TL;DR: I think we should probably avoid it.

I did a similar thing for the "already existing processor name" error. As soon as the user changes the name, the error is gone. I still have mixed feelings from time to time when I see this behavior and I question it in my mind. 

When changing the name we are sure that the error about the previous name will be gone. Still, the user could pick another name already taken (how unlucky), triggering again the error after submitting.

We could say this is an "optimistic approach". In its defense we know that once the name is changed, you can try again and take another shot with a chance of picking a valid one.

About TT, the situation is similar but there's a difference. When changing the TT, we don't know if the change has any chance of solving the issue. Maybe the problem was caused by a missing bracket. If I just rename a variable instead, there's no chance the error will go away. It's difficult then to keep the same "optimistic approach".

I know this could look inconsistent with the "resource name validation" treatment. I am not sure if the difference between the two is big enough to justify two different strategies. Anyway, we could also decide to remove the _clean-on-change_  behavior from the processor/bridge name too.

@vpellegrino @manstis What do you think? I'm curious to hear from you.

Reply:
Thanks for raising this interesting point @kelvah 
I reasoned about it while developing, and I concluded that from a user perspective, the validation should be consistent for all fields, except if there is a valid reason.
However, for this field, a new error will likely be raised after clicking on submit, so I will avoid clearing the TT error on template change.